### PR TITLE
Feature/use storage hook

### PR DESCRIPTION
**Pull request by Christiantyemele**

### CODE REVIEW

**Comment by Christiantyemele** [comment]
@stephane-segning  requesting for review on pull request

**Comment by stephane-segning** [comment]
Which ticket is it related to please @Christiantyemele ?

File: docs/useStorage.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#             useStorage()  HOOK
```
Comment (stephane-segning) [inline docs/useStorage.md:1]: This first part explain how a react hook is working. It's not required.

File: power-pay-frontend/src/Hooks/useStorage.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    const storageValue = storage?.getItem(key);
+
+    if (storageValue != null) return JSON.parse(storageValue);
+
+    return defaultValue;
+  });
+  // here we use the useEffect hook to check if the value is present hence we remove the else we update the value
+
+  useEffect(() => {
+    if (value === undefined) return storage?.removeItem(key);
```
Comment (stephane-segning) [inline power-pay-frontend/src/Hooks/useStorage.ts:33]: So, we delete everything when we unmount? Why?

File: power-pay-frontend/src/Hooks/useStorage.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import { useState, useEffect, useCallback } from "react";
+
+// we define our hook variable specifying its paramaters and types and also the storing logic
+const useStorage = (
+  key: string,
+  defaultValue: number | string | null = null,
+  type?: "local" | "session"
+) => {
+  const storage =
```
Comment (stephane-segning) [inline power-pay-frontend/src/Hooks/useStorage.ts:11]: What if I want to save into the `indexedDb`? For the requirements of this app, we won't use use sessionStorage. For more sensible things, we'll use indexedDb and localStorage for the rest

File: power-pay-frontend/src/Tests/useStorageTest.tsx
Comment (stephane-segning) [inline power-pay-frontend/src/Tests/useStorageTest.tsx:1]: What is a test file doing in `src/`? Why and this test is doing nothing actually

**Review by stephane-segning — CHANGES_REQUESTED** [review]
This won't work. We want a storage system capable of being used everywhere in our app. For that, we need to leverage the capabilities of react, by using react context or redux. You should document that before writing code. Plus, your test file is not explicit enough (it's just an example – you could do that in the doc). Remove that too.

**Comment by Christiantyemele** [comment]
Hello sir @stephane-segning i have updated the useStorage docs and requesting for review before proceeding.

File: power-pay-frontend/src/Hooks/useStorage.ts
Comment (stephane-segning) [inline power-pay-frontend/src/Hooks/useStorage.ts:1]: Remove this file

File: power-pay-frontend/src/Tests/useStorageTest.tsx
Comment (stephane-segning) [inline power-pay-frontend/src/Tests/useStorageTest.tsx:1]: Remove this file

File: docs/useStorage.md
Comment (stephane-segning) [inline docs/useStorage.md:1]: You described well how react context and redux are working. But not how the `useStorage` should work

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Please work on these points

**Comment by Christiantyemele** [comment]
@stephane-segning have done requested changes and requesting review

File: docs/useStorage.md [hunk 4]
```diff
@@ -1,46 +1,51 @@
+
 # useStorage Hook
 
  The react useStorage hook is a react hook which allows use to easily manage state in a web storage API like a localstorage or sessionstorage.<br>
```
Comment (stephane-segning) [inline docs/useStorage.md:4]: It's not a react hook. It is a hook written in react, but it's not coming from react

File: docs/useStorage.md [hunk 5]
```diff
@@ -1,46 +1,51 @@
...
+
 The **store**: An immutable object that holds the application's state<br>
 **Reducer**: A function that returns some data, triggered by an action type<br>
 **Action**: An object that tells the reducer how to change the state it must contain a type property and optional payload property.<br>
 Points to note are.<br>
-#
-- Redux is typically used for lager-scale management needs or for applications with complex state logic that require features such as as middleware, time-travel debigging and predictable container state.
 
-To make the most use of the useStorage hook we are going to be using the Context library here this is because of its simplicity, also the Context api is a build in feature of react, the Context is suitable for managing global state or application-level state and also for the scope of our applications since we donot have to much data to handle.<br>
+- Redux is typically used for lager-scale management needs or for applications with complex state logic that require features such as as middleware, time-travel debigging and predictable container state.<br>
```
Comment (stephane-segning) [inline docs/useStorage.md:43]: Why are even talking about redux anyway?

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Please focus this only on the documentation of how the useStorage should work, nothing more. Avoid any unuseful comments about react context or redux. If you plan using react context, then explain how to use it to implement the hook.

File: docs/useStorage.md (hunk 5 above)
Comment (Christiantyemele) [reply docs/useStorage.md:43]: talking about Context and redux is to justify the choice of using Context instead of redux sir @stephane-segning

**Comment by Christiantyemele** [comment]
@stephane-segning have added visual flow to docs

File: docs/useStorage.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+The usestorage hook performs its task by taking the key and associated value converting it to a json string then using the ```setItem()``` method to store it in the storage with its associated key. And also has a ```getItem()``` for retrieving the value stored based on the corresponding key and a ```removeItem()``` method which will be used to remove key value pair from the storage. <br>
+To conclude the react Context and useStorage helps our application to have reusable and persist data accross components.<br>
+The flow is as shown
+
+```mermaid
+%%{init: {'theme': 'base', 'themeVariables': { 'componentBackground': '#EEEEEE', 'componentBorder': '#CCCCCC', 'componentTextColor': '#333333'}}}%%
```
Comment (stephane-segning) [inline docs/useStorage.md:29]: Remove all these styling

**Review by stephane-segning — CHANGES_REQUESTED** [review]

**Comment by Christiantyemele** [comment]
@AssahBismarkabah  please can you check useStorage docs if clear enough

**Pull request merged**